packages/notification-service/src/invites/invites.ts

```diff
@@ -7,27 +8,47 @@ import { getContractKit } from '../util/utils'
 
 const TAG = 'INVITES'
 
-export async function handleInvites() {
-  const kit = await getContractKit()
-  const fromBlock = getLastInviteBlockNotified() + 1
-  if (fromBlock <= 0) {
+let sendingInvites = false
+
+function updateLastInviteBlock(fromBlock: number, events: EventLog[]) {
+  if (events.length === 0) {
     return
   }
-  console.debug(TAG, `Starting to fetch invites from block ${fromBlock}`)
+  const maxBlock = events
+    .map((event) => event.blockNumber)
+    .reduce((max, current) => Math.max(max, current), fromBlock)
+  setLastInviteBlockNotified(maxBlock)
+}
 
-  const escrow = await kit.contracts.getEscrow()
-  const events = await escrow.getPastEvents('Withdrawal', {
-    fromBlock,
-  })
-  console.debug(TAG, `Got ${events.length} escrow withdrawal events`)
-  let maxBlock = fromBlock
-  for (const event of events) {
-    maxBlock = Math.max(event.blockNumber, fromBlock)
-    const inviter = event.returnValues.to.toLowerCase()
-    console.debug(TAG, `Sending invite notification to ${inviter}`)
-    await sendInviteNotification(inviter)
-  }
-  if (maxBlock > fromBlock) {
-    setLastInviteBlockNotified(maxBlock)
+export async function handleInvites() {
+  try {
+    if (sendingInvites) {
+      return
+    }
+    sendingInvites = true
```

Comment:
I think this isn't needed, last time I checked, `AsyncPolling` is not scheduling a new run until the last run finished.

Reply:
You're right. I wasn't too sure about that, I'll remove that part from the description of the PR as well